sdlib/d/gc/extent.d

```diff
@@ -352,13 +361,40 @@ public:
 	@property
 	ulong usedCapacity() {
 		assert(isLarge(), "usedCapacity accessed on non large!");
-		return _metadata.usedCapacity;
+		return _metadata.largeData.usedCapacity;
 	}
 
 	void setUsedCapacity(size_t size) {
 		assert(isLarge(), "Cannot set used capacity on a slab alloc!");
-		_metadata.usedCapacity = size;
+		_metadata.largeData.usedCapacity = size;
 	}
+
+	@property
+	void function(void* ptr, size_t size) finalizer() {
+		assert(isLarge(), "Finalizer accessed on non large!");
+		return cast(void function(void* ptr, size_t size))
+			_metadata.largeData.finalizer;
+	}
+
+	void setFinalizer(void* finalizer) {
+		assert(isLarge(), "Cannot set finalizer on a slab alloc!");
+		_metadata.largeData.finalizer = finalizer;
+	}
```

Comment:
I don't understand what's up here.

It's like you put one foot forward and then don't realize that the second foot can move forward too.

If finalizer gives you a function, then setFinalizer should accept a fucntion, and the field itself almost certainly would benefit from being a function.